### PR TITLE
Pv2 855 data locks load apprenticeships into the cache

### PR DESCRIPTION
**Pull request by samuel-dev-dba**

### CODE REVIEW

File: src/SFA.DAS.Payments.Application/Data/Configurations/ApprenticeshipModelConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using SFA.DAS.Payments.Model.Core.Entities;
+
+namespace SFA.DAS.Payments.Application.Data.Configurations
+{
+    class ApprenticeshipModelConfiguration : IEntityTypeConfiguration<ApprenticeshipModel>
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.Application/Data/Configurations/ApprenticeshipModelConfiguration.cs]: needs a class modifier

File: src/SFA.DAS.Payments.Application/Data/Configurations/ApprenticeshipPriceEpisodeModelConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using SFA.DAS.Payments.Model.Core.Entities;
+
+namespace SFA.DAS.Payments.Application.Data.Configurations
+{
+    class ApprenticeshipPriceEpisodeModelConfiguration : IEntityTypeConfiguration<ApprenticeshipPriceEpisodeModel>
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.Application/Data/Configurations/ApprenticeshipPriceEpisodeModelConfiguration.cs]: needs a class modifier

File: src/SFA.DAS.Payments.Application/Data/Configurations/ApprenticeshipModelConfiguration.cs (hunk 1 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.Application/Data/Configurations/ApprenticeshipModelConfiguration.cs]: done

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndStepsBase.cs [hunk 3]
```diff
@@ -177,67 +165,92 @@ protected void AddTestAims(IList<Aim> aims, long ukprn)
...
-            foreach (var mappedCommitment in mappedCommitments)
+            if (apprenticeshipSpec.Ukprn == default(long))
             {
-                var matchedCommitment =
-                    await DataContext.Commitment.FirstOrDefaultAsync(e =>
-                        e.CommitmentId == mappedCommitment.CommitmentId &&
-                        e.VersionId == mappedCommitment.VersionId);
-
-                if (matchedCommitment != null)
+                if (string.IsNullOrEmpty(apprenticeshipSpec.Provider))
```
[the remark this answers is not in the document]
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndStepsBase.cs]: done

File: src/SFA.DAS.Payments.Application/Data/Configurations/ApprenticeshipPriceEpisodeModelConfiguration.cs (hunk 2 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.Application/Data/Configurations/ApprenticeshipPriceEpisodeModelConfiguration.cs]: done

**Pull request merged**